src/main/java/hu/elte/eserial/mapper/MapMapper.java

```diff
@@ -0,0 +1,43 @@
+package hu.elte.eserial.mapper;
+
+import hu.elte.eserial.exception.EserialMapperMismatchException;
+import hu.elte.eserial.util.TypeUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Maps Map-like objects (e.g Map, HashMap).
+ */
+public class MapMapper implements ObjectMapper {
+
+    /**
+     * Returns the mapped representation of the map {@code that}.
+     *
+     * @param that a map
+     * @return mapped representation of {@code that}
+     */
+    @Override
+    public Object map(Object that) {
+        if (!TypeUtils.isMap(that.getClass())) {
+            throw new EserialMapperMismatchException(that.getClass());
+        }
+
+        Map map = (Map) that;
+
+        return map
+                .entrySet()
+                .stream()
+                .map(entryObject -> {
+                    Map.Entry entry = (Map.Entry) entryObject;
+
+                    Map<String, Object> entryMap = new HashMap<>();
```

Comment:
It would be nice to distinguish between "serializable" and "compound" keys. For the latter this approach is fine, but if the key is of a serializable type ("inline serializable", like a String or a number), the input Map could be returned as is.

Reply:
+1

Also yes, the values should be mapped.

Reply:
According to @hegedusadam this is the expected behaviour.